src/nih_wayfinding/test/spec/routing/abbreviate-filter.spec.js

```diff
@@ -0,0 +1,51 @@
+
+'use strict';
+
+describe('nih.routing: Filters Abbreviate', function () {
+
+    beforeEach(module('nih.routing'));
+
+    var filter;
+
+    var addressCase1 = 'North Baltimore Avenue';
+    var addressWhitespace = '  North  \n  Baltimore    \t Avenue     ';
+    var addressCase2 = 'Baltimore Avenue North';
+    var addressCase3 = 'Baltimore Avenue';
+
+    beforeEach(inject(function ($filter) {
+        filter = $filter;
+    }));
+
+    it('should return same as input on bad input', function () {
+        var result = filter('abbreviate')(null);
+        expect(result).toEqual(null);
+
+        result = filter('abbreviate')(1000);
+        expect(result).toEqual(1000);
+
+        result = filter('abbreviate')('');
+        expect(result).toEqual('');
+    });
+
+    it('should handle extra whitespace', function () {
+        var result = filter('abbreviate')(addressWhitespace);
+        expect(result).toEqual('N Baltimore Ave');
+    });
+
+    it('should handle the standard cases', function () {
```

Comment:
Might be good to test for 'S South Something Street', to make sure 'South' doesn't get abbreviated twice, since we have one of those in our area.